src/crafting.cpp

```diff
@@ -1625,12 +1670,12 @@ void player::complete_craft()
     // If we're here, the craft was a success!
     // Use up the components and tools
     std::list<item> used;
-    for (const auto &it : making->requirements.components) {
-        std::list<item> tmp = consume_items(it, batch_size);
+    for (int i = 0; i < making->requirements.components.size(); i++) {
+        std::list<item> tmp = consume_items(comp_selections[i], batch_size); // again, we consume the items we selected earlier
         used.splice(used.end(), tmp);
     }
-    for (const auto &it : making->requirements.tools) {
-        consume_tools(it, batch_size);
+	for (int i = 0; i < making->requirements.tools.size(); i++) {
```

Comment:
good point, I first had the idea of some mapping of the selected component to the component group, which didn't work to well, but it flowed into this form.